src/libexpr/lexer.l

```diff
@@ -160,6 +163,33 @@ or          { return OR_KW; }
               }
               return INT;
             }
+{BIN}       { errno = 0;
+              yylval->n = strtoll(yytext + 2, 0, 2);
+              if (errno != 0)
+                  throw ParseError({
+                      .msg = hintfmt("invalid binary integer '%1%'", yytext),
+                      .errPos = data->state.positions[CUR_POS],
+                  });
```

Comment:
Could you test these exceptions? They're triggered when we run out of bits, for example:


```

nix-repl> -9223372036854775808
error: invalid integer '9223372036854775808'

       at «string»:1:2:

            1| -9223372036854775808
             |  ^

```

Reply:
Where should I write these tests ? Is there a particular test section in the source related to the lexer ?

Reply:
I think you could add it to `tests/lang.sh` by adding a `parse-fail-*.nix` file.